MdeModulePkg/Core/Dxe/Mem/Page.c

```diff
@@ -141,6 +151,71 @@ RemoveMemoryMapEntry (
   }
 }
 
+/**
+  Get the memory type for a given bucket.
+
+  @param  PhysicalStart  The starting address of the memory region.
+  @param  PhysicalEnd    The ending address of the memory region.
+
+  @return The memory type for the bucket that contains the given physical address range.
+          If the address range does not match any special bucket, it returns EfiMaxMemoryType.
+**/
+EFI_MEMORY_TYPE
+GetBucketMemoryType (
+  IN EFI_PHYSICAL_ADDRESS  PhysicalStart,
+  IN EFI_PHYSICAL_ADDRESS  PhysicalEnd
+  )
+{
+  EFI_MEMORY_TYPE  BucketType;
+
+  // Find the bucket type for the incoming memory region.
+  for (BucketType = 0; BucketType < EfiMaxMemoryType; BucketType++) {

```

Comment:
An explicit cast might be needed to avoid a warning for inconsistent type assignment in some compilers.

```suggestion
  for (BucketType = (EFI_MEMORY_TYPE)0; BucketType < EfiMaxMemoryType; BucketType++) {
```